src/libraries/System.Text.Json/tests/System.Text.Json.FSharp.Tests/EnumTests.fs

```diff
@@ -0,0 +1,51 @@
+﻿module System.Text.Json.Tests.FSharp.EnumTests
+
+open System
+open System.Text.Json
+open System.Text.Json.Serialization
+open Xunit
+
+[<Flags>]
+type BadEnum =
+  | ``There's a comma, in my name`` = 1
+  | ``There's a comma, even here`` = 2
+  | ``ThisisagoodEnumValue`` = 4
+
+let badEnum = BadEnum.``There's a comma, in my name`` ||| BadEnum.``There's a comma, even here``
+let badEnumJsonStr = @$"""{badEnum.ToString()}"""
+
+let badEnumWithGoodValue = BadEnum.``ThisisagoodEnumValue``
+let badEnumWithGoodValueJsonStr = @$"""{badEnumWithGoodValue.ToString()}"""
+
+[<Flags>]
+type GoodEnum =
+  | ``Thereisnocommainmyname_1`` = 1
+  | ``Thereisnocommaevenhere_2`` = 2
+
+let goodEnum = GoodEnum.``Thereisnocommainmyname_1`` ||| GoodEnum.``Thereisnocommaevenhere_2``
+let goodEnumJsonStr = @$"""{goodEnum.ToString()}"""
+
+let options = new JsonSerializerOptions()
+options.Converters.Add(new JsonStringEnumConverter())
+
+[<Fact>]
+let ``Deserialize With Exception If Enum Contains Special Char`` () =
+    Assert.Throws<JsonException>(fun () -> JsonSerializer.Deserialize<BadEnum>(badEnumJsonStr, options) |> ignore)
+
+[<Fact>]
+let ``Serialize With Exception If Enum Contains Special Char`` () =
+    Assert.Throws<JsonException>(fun () ->  JsonSerializer.Serialize(badEnum, options) |> ignore)
+
+[<Fact>]
+let ``Successful Deserialize Normal Enum`` () =
+    let actual = JsonSerializer.Deserialize<GoodEnum>(goodEnumJsonStr, options)
+    Assert.Equal(GoodEnum.Thereisnocommainmyname_1 ||| GoodEnum.Thereisnocommaevenhere_2, actual)
+
+[<Fact>]
+let ``Success Deserialize Good Value Of Bad Enum Type`` () =
```

Comment:
The last two tests seem kind of inconsistent. If deserialization works then I would expect serialization to work. Alternatively, if one operation fails then both should fail. 

I think though that the only way we could guarantee partial support for well-formed enums is by including a check for special characters on every input string. Consequently, I think we should instead have serialization and deserialization _always_ fail when `s_containsSpecialChars` is true. This could be done in the converter constructor in order to avoid adding another branch in the `Write` and `Read` methods.

Reply:
> I think though that the only way we could guarantee partial support for well-formed enums is by including a check for special characters on every input string

I think we can add a contains special char tag for every enum value. So that we can successful serialize well-formed enums. This is consistent with Enum.TryParse behavior in F#.